smk/gene_abundance.smk

```diff
@@ -452,18 +422,26 @@ rule merge_msamtools_genome_mapping_profiles:
                                             sample = ilmn_samples)
     output:
         combined="{wd}/{omics}/9-mapping-profiles/{post_analysis_out}/all.p{identity}.profile.{type}.txt"
+    params:
+        files = lambda wildcards, input: ",".join(input.single)
```

Comment:
If previously run was used to avoid the shell character limits, joining them like this could result in similar problem. PErhaps the R script can find the inputs in "{wd}/{omics}/9-mapping-profiles/{post_analysis_out}/", as all {sample} folders would be inlcuded.

Reply:
Good point. The difference here is that number of samples rarely goes above 1000, but number of genomes (where we had to move to `run:`) very easily crosses 5000. Also, I did not quite identify the actual problem with the other issue where I had to switch to `run:`. I definitely did not cross bash limits, but it stopped working as I increased sample count. I gave up debugging and just reimplemented and got identical output.

I prefer to have the input files explicitly names to the script rather than it evaluating the directory. E.g., if we want to test something quickly on only 10 out of 1000 samples by changing the `ILLUMINA:` field, then it will have no effect as it will evaluate all 1000 samples within the folder. Best choice will be to switch to `run:` but we need to run an R script here so it is unavoidable. I hope nobody runs MIntO on 100,000 samples.

And I use the string only once here. In the other example, I had to use the list twice, which made it tricky.